itest/lnd_multi-hop_test.go

```diff
@@ -1956,48 +1959,58 @@ func runMultiHopHtlcAggregation(ht *lntest.HarnessTest,
 		lnrpc.CommitmentType_SIMPLE_TAPROOT:
 
 		// Bob should have `numInvoices` for both HTLC success and
-		// timeout txns, plus one anchor sweep.
-		ht.AssertNumPendingSweeps(bob, numInvoices*2+1)
+		// timeout txns.
+		ht.AssertNumPendingSweeps(bob, numInvoices*2)
 
 		// Carol should have commit and anchor outputs.
 		ht.AssertNumPendingSweeps(carol, 2)
 
-		// We expect to see three sweeping txns:
-		// 1. Bob's sweeping tx for all timeout HTLCs.
+		// We expect to see one sweeping txns:
 		// 2. Bob's sweeping tx for all success HTLCs.
-		// 3. Carol's sweeping tx for her commit and anchor outputs.
-		expectedTxes = 3
+		expectedTxes = 1
 
 	default:
 		ht.Fatalf("unhandled commitment type %v", c)
 	}
 
-	// Mine a block to confirm Bob's anchor sweeping, which will also
-	// trigger his sweeper to sweep HTLCs.
-	ht.MineBlocksAndAssertNumTxes(1, 1)
-
-	// Assert the sweeping txns are found in the mempool.
-	txes := ht.Miner.GetNumTxsFromMempool(expectedTxes)
+	// Mine the block with the following transactions.
+	// 1. Carol's sweeping tx for her commit and anchor outputs.
+	// 2. Bob's sweeping tx for all timeout HTLCs.
+	// This block will also trigger the htlc-success sweep transaction.
+	minedTx := ht.MineBlocksAndAssertNumTxes(1, 2)
+	require.Len(ht, minedTx, 1)
 
-	// Since Bob can aggregate the transactions, we expect a single
-	// transaction, that have multiple spends from the commitment.
 	var (
 		timeoutTxs []*chainhash.Hash
 		successTxs []*chainhash.Hash
 	)
-	for _, tx := range txes {
+
+	for _, tx := range minedTx[0].Transactions {
 		txid := tx.TxHash()
 
 		for i := range tx.TxIn {
 			prevOp := tx.TxIn[i].PreviousOutPoint
-			if _, ok := successOuts[prevOp]; ok {
-				successTxs = append(successTxs, &txid)
+			if _, ok := timeoutOuts[prevOp]; ok {
+				timeoutTxs = append(timeoutTxs, &txid)
 
 				break
 			}
+		}
+	}
```

Comment:
The outer loop here is a `fn.Filter` and the inner one is a `fn.Find`

Reply:
code change is obsolete